website/addons/googledrive/tests/test_models.py

```diff
@@ -131,296 +131,362 @@ def test_node_addon_get_or_create_finds_changed_root(self):
-
-
-class TestGoogleDriveNodeSettingsModel(OsfTestCase):
+        self.external_account = ExternalAccountFactory()
 
-    def setUp(self):
-        super(TestGoogleDriveNodeSettingsModel, self).setUp()
-        self.user = UserFactory()
-        self.user.add_addon('googledrive')
+        self.user.external_accounts.append(self.external_account)
         self.user.save()
-        self.user_settings = self.user.get_addon('googledrive')
-        oauth_settings = GoogleDriveOAuthSettingsFactory()
-        oauth_settings.save()
-        self.user_settings.oauth_settings = oauth_settings
+
+        self.user_settings = self.user.get_or_add_addon('googledrive')
+
+    def test_grant_oauth_access_no_metadata(self):
+        self._prep_oauth_case()
+
+        self.user_settings.grant_oauth_access(
+            node=self.node,
+            external_account=self.external_account,
+        )
         self.user_settings.save()
-        self.project = ProjectFactory()
-        self.node_settings = GoogleDriveNodeSettingsFactory(
-            user_settings=self.user_settings,
-            owner=self.project,
+
+        assert_equal(
+            self.user_settings.oauth_grants,
+            {self.node._id: {self.external_account._id: {}}},
         )
 
-    def test_fields(self):
-        node_settings = GoogleDriveNodeSettings(user_settings=self.user_settings)
-        node_settings.save()
+    def test_grant_oauth_access_metadata(self):
+        self._prep_oauth_case()
 
-        assert_true(node_settings.user_settings)
-        assert_true(hasattr(node_settings, 'folder_id'))
-        assert_true(hasattr(node_settings, 'folder_path'))
-        assert_true(hasattr(node_settings, 'folder_name'))
-        assert_equal(node_settings.user_settings.owner, self.user)
+        self.user_settings.grant_oauth_access(
+            node=self.node,
+            external_account=self.external_account,
+            metadata={'folder': 'fake_folder_id'}
+        )
+        self.user_settings.save()
 
-    def test_complete_true(self):
-        assert_true(self.node_settings.has_auth)
-        assert_true(self.node_settings.complete)
+        assert_equal(
+            self.user_settings.oauth_grants,
+            {
+                self.node._id: {
+                    self.external_account._id: {'folder': 'fake_folder_id'}
+                },
+            }
+        )
 
-    def test_complete_false(self):
-        self.node_settings.folder_id = None
+    def test_verify_oauth_access_no_metadata(self):
+        self._prep_oauth_case()
 
-        assert_true(self.node_settings.has_auth)
-        assert_false(self.node_settings.complete)
+        self.user_settings.grant_oauth_access(
+            node=self.node,
+            external_account=self.external_account,
+        )
+        self.user_settings.save()
 
-    def test_complete_auth_false(self):
-        self.node_settings.user_settings = None
+        assert_true(
+            self.user_settings.verify_oauth_access(
```

Comment:
For readability, it might help to separate these statements:
```
has_access = self.user_settings.verify_oauth_access  ...  
assert_true(has_access)
```